core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1704,6 +1705,44 @@ private void analyzeFrameRangeOffset(Expression offsetValue, FrameBound.Type bou
             frameBoundCalculations.put(NodeRef.of(offsetValue), function);
         }
 
+        @Override
+        protected Type visitStaticMethodCall(StaticMethodCall node, Context context)
+        {
+            // PostgreSQL-style are syntactically ambiguous with static method calls. So, static method call semantics take precendence.
+            // A static method call is characterized by the target being an expression whose type is "type". This not yet supported
+            // as a first-class concept, so we fake it by analyzing the expression normally. If the analysis succeeds, we treat it as
+            // the target of a cast. If the analysis fails, check whether the target is an identifier matching a known type name.
```

Comment:
What should be the result of 
```
SELECT bigint::real FROM a_table_that_has_column_named_bigint
```

i.e. treating lhs as type name takes precedence, or treating it as an expression does?

Reply:
That should fail because it looks like a static method call (real method with no args on bigint type). It needs to behave like that for forward compatibility.